types/recoil/lib/core/state.d.ts

```diff
@@ -0,0 +1,23 @@
+import { Loadable } from './loadable';
+
+export type NodeKey = string;
```

Comment:
Is there any way to ensure that this value is globally unique?

Reply:
I think that's not viable on multiple fronts:

1. TS doesn't narrow primitive literals: https://github.com/microsoft/TypeScript/issues/16896
2. Even if it did, it wouldn't be able to do that for dynamically constructed IDs (which is an explicit feature of recoil)
3.  And even if those two were true, uniqueness must only hold within a given recoil context. I don't think checking for the presence of a given react context is even on the far horizon

Which is all unfortunate, if indeed true, cause I like the idea

Reply:
well said Gregoor! Unfortunately uniqueness can only be enforced at runtime for the time being (an error will be thrown)